hazelcast/src/main/java/com/hazelcast/internal/metrics/ProbeBuilder.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.metrics;
+
+/**
+ * Immutable builder object to register Probes.
+ */
+public interface ProbeBuilder {
+
+    /**
+     * Returns a new ProbeBuilder instance with the given tag added.
+     */
+    ProbeBuilder withTag(String tag, String value);
+
+    /**
+     * Builds the String name for manually registering probes.
+     */
+    String metricName();
+
+    /**
+     * Registers a single probe.
+     * <p>
+     * If a probe for the given name exists, it will be overwritten silently.
+     *
+     * @param source the object to pass to probeFn
+     * @param metricName the value of "metric" tag
+     * @param level the ProbeLevel
+     * @param probeFn the probe function
+     * @throws NullPointerException if any of the arguments is null
+     */
+    <S> void register(S source, String metricName, ProbeLevel level, DoubleProbeFunction<S> probeFn);
```

Comment:
I would drop the following 2 methods. They are low level methods.

Reply:
We use them in Jet. I double checked, but it's not trivial to replace. It's available next to `scanAndRegister` in `MetricsRegistry`, so the same is in `ProbeBuilder`.

Reply:
ok